vmdk_plugin/plugin.go

```diff
@@ -208,11 +209,50 @@ func (d *vmdkDriver) unmountVolume(name string) error {
 // (until Mount is called).
 // Name and driver specific options passed through to the ESX host
 func (d *vmdkDriver) Create(r volume.Request) volume.Response {
-	err := d.ops.Create(r.Name, r.Options)
+	// Defaults to ext4 if not specified
+	if _, ok := r.Options["fstype"]; ok == false {
+		r.Options["fstype"] = "ext4"
+	}
+
+	// Verify the existance of filesystem tools
+	mkfscmd := "/sbin/mkfs." + r.Options["fstype"]
+	_, err := os.Lstat(mkfscmd)
+	if err != nil {
+		log.WithFields(log.Fields{"name": r.Name, "error": err}).Error("Not found ")
+		return volume.Response{Err: err.Error()}
+	}
+
+	err = d.ops.Create(r.Name, r.Options)
 	if err != nil {
 		log.WithFields(log.Fields{"name": r.Name, "error": err}).Error("Create volume failed ")
 		return volume.Response{Err: err.Error()}
 	}
+
+	// Handle filesystem creation
+	dev, err := d.ops.Attach(r.Name, nil)
+	if err != nil {
+		log.WithFields(log.Fields{"name": r.Name, "error": err}).Error("Attach volume failed ")
+		return volume.Response{Err: err.Error()}
+	}
+
+	device, err := fs.GetDevicePath(dev)
+	if err != nil {
+		log.WithFields(log.Fields{"name": r.Name, "error": err}).Error("Could not find attached device ")
+		return volume.Response{Err: err.Error()}
+	}
+
+	err = fs.Mkfs(mkfscmd, r.Name, device)
```

Comment:
Note that the current code would not do well with the slow format either. In fact, it will block all requests on a single ESX. This change already improves this so that only a single VM is impacted - even without async or printing progress.  So it's all good , just add the logs please.